docs/website/docs/developers/debugging/gpu.md

```diff
@@ -0,0 +1,234 @@
+---
+hide:
+  - tags
+tags:
+  - GPU
+  - CUDA
+  - Metal
+  - ROCm
+  - Vulkan
+icon: octicons/bug-16
+---
+
+# GPU debugging playbook
+
+This page aims to provide general approaches and practical tips for debugging
+GPU compiler/runtime correctness/performance issues in IREE.
+
+GPUs fundamentally have similar architectures and software stacks.
+We target GPUs from various vendors using different GPU APIs, but they share
+quite a lot common infrastructure in IREE.
+So the approaches and tips here should be widely applicable.
+
+For the ones that are specific to a particular kind of problem/component/GPU,
+they are prefixed with proper icons to be clear.
+Here are what those icons represents--
+
+Icon | Category
+:--: | :------:
+:material-check-bold: | Correctness
+:material-trending-up: | Performance
+:simple-amd: | AMD HIP/ROCm
+:simple-apple: | Apple Metal
+:simple-microsoft: | Microsoft DirectX
+:simple-nvidia: | NVIDIA CUDA
+:simple-vulkan: | Vulkan
```

Comment:
Interesting idea, but I'd just inline the category text. instead of requiring a legend lookup, at least in the admonition titles.

![image](https://github.com/openxla/iree/assets/4010439/82eb3ddc-9276-4b5f-bbb4-62f58a740d34)